File: SlidingWindow/3_longest_substring/main.go
package main

func lengthOfLongestSubstring(s string) int {
	var length, left int
	state := map[string]bool{}
	for idx, elem := range s {
		inner:
		for {
			ok := state[string(elem)]
			if ok{
				delete(state, string(s[left]))
				left++
				continue
			}
			break inner
		}
		state[string(elem)] = true
		if length < idx-left+1 {
			length = idx - left + 1
		}
	}
	return length
}

File: SlidingWindow/3_longest_substring/main.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        length, left, state = 0, 0, set()
        for idx, elem in enumerate(s):
            while elem in state:
                state.remove(s[left])
                left += 1
            state.add(elem)
            length = max(length, idx - left +1)

        return length
